includes/shell/grammar/grammar_util.hpp

```diff
@@ -16,14 +16,23 @@ struct GrammarUtil {
 	}
 
 	template <Token::Type TOKEN_TYPE>
-	static optional<Ast::Node> ConsumeIf(TokenProvider &tokens) {
-		auto token = tokens.peek();
-		if (token->getType() != TOKEN_TYPE) {
-			return nullopt;
+	struct ConsumeIf {
+	public:
+		static Rule make() {
+			return Rule {[]() -> vector<Rule::Option> {
+				return {Rule::Terminal {ConsumeIf<TOKEN_TYPE>::OP}};
+			}};
 		}
-		tokens.consume();
-		return Ast::Literal(std::move(token.value()));
-	}
+
+	public:
+		static optional<Ast::Node> OP(TokenProvider &tokens) {
```

Comment:
What does `OP` stand for?

Reply:
Operation ? Idk this is mostly our convention at work for functors